experiments/multichain/multichain_client.py

```diff
@@ -0,0 +1,159 @@
+        target = self.all_vars[candidate_id]
+        peer = (str(target['host']), target['port'])
+        self.scheduler.update_amount_send(peer, int(amount_sent) * 1000)
+
+    def increase_kbytes_received(self, candidate_id, amount_received):
+        """
+        Increase the Kbytes received from a particular node in the MultiChainScheduler.
+        """
+        msg("%s: Increasing bytes received: %s For candidate: %s" % (self.my_id, amount_received, candidate_id))
+        target = self.all_vars[candidate_id]
+        peer = (str(target['host']), target['port'])
+        self.scheduler.update_amount_received(peer, int(amount_received)*1000)
+
+    def schedule_block(self, candidate_id):
+        """
+        Finish up the download.
+        :param candidate_id:
+        """
+        msg("%s: Scheduling block for candidate: %s" % (self.my_id, candidate_id))
+        target = self.all_vars[candidate_id]
+        peer = (str(target['host']), target['port'])
+        self.scheduler.schedule_block(peer)
+
+    def introduce_candidates(self):
+        """
+        Introduce every candidate to each other so that later the candidates can be retrieved and used as a destination.
+        """
+        msg("Introducing every candidate")
+        for node in self.all_vars.itervalues():
+            candidate = Candidate((str(node['host']), node['port']), False)
+            self._community.add_discovered_candidate(candidate)
+
+    @property
+    def my_member_key_curve(self):
+        return u"curve25519"
+
+    def close(self):
+        msg("close command received")
+        self._community.unload_community()
+
+
+class MultiChainDelayCommunity(MultiChainCommunity):
+    """
+    Test Community that delays signature requests.
+    """
+    delay = 3
+
+    def __init__(self, *args, **kwargs):
+        super(MultiChainDelayCommunity, self).__init__(*args, **kwargs)
+
+    def allow_signature_request(self, message):
+        """
+        Ignore the signature requests.
+        :param message: the to be delayed request
+        """
+        self.logger.info("Received signature request that will delayed for %s." % self.delay)
+        sleep(self.delay)
+        self.logger.info("Delay over.")
+        super(MultiChainDelayCommunity, self).allow_signature_request(message)
+
+
+class MultiChainNoResponseCommunity(MultiChainCommunity):
+    """
+    Test Community that does not respond to signature requests.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(MultiChainNoResponseCommunity, self).__init__(*args, **kwargs)
+
+    def allow_signature_request(self, message):
+        """
+        Ignore the signature requests.
+        :param message: the to be ignored request
+        """
+        self.logger.info("Received signature request that will be ignored.")
+        return
+
+if __name__ == '__main__':
+    MultiChainClient.scenario_file = environ.get('SCENARIO_FILE', 'multichain_standalone.scenario')
+    main(MultiChainClient)
```

Comment:
missing end of the line. I suggest you do an autopep8 pass to all the files you have created.